merlin-lib.opam

```diff
@@ -0,0 +1,26 @@
+opam-version: "2.0"
+maintainer:   "defree@gmail.com"
+authors:      "The Merlin team"
+homepage:     "https://github.com/ocaml/merlin"
+bug-reports:  "https://github.com/ocaml/merlin/issues"
+dev-repo:     "git+https://github.com/ocaml/merlin.git"
+license:      "MIT"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "4.14" & < "4.15"}
+  "dune" {>= "2.9.0"}
+  "yojson" {>= "1.6.0"}
```

Comment:
Why does the backend need json encoding/decoding? Would it be possible to move this dependency to the frontend?

Reply:
There are dumping functions in `Query_commands` (and `Mconfig` and `Warnings`) that produce Json.
The `Dump` query is the only one that produces json before going through `Query_json.of_response`.

@let-def is there a reason for that ? It looks like it could be refactored but it is probably a bit tedious to do.

Reply:
OCaml code is annoying to debug and having a common format for printing internals helped in the past.
However backend and frontend only needs to agree on this format.
To remove the dependency on Yojson, we can just copy the polymorphic variant definition of json in the backend.